test/mlir/onnx/onnx_constprop.mlir

```diff
@@ -103,8 +103,9 @@ func.func @test_add_constant_4(%arg0 : tensor<3xi32>) -> tensor<3xi32> {
   %1 = onnx.Constant dense<[10, 11, 12]> : tensor<3xi32>
   %2 = "onnx.Add"(%0, %arg0) : (tensor<3xi32> , tensor<3xi32>) -> tensor<3xi32>
   %3 = "onnx.Add"(%1, %2) : (tensor<3xi32> , tensor<3xi32>) -> tensor<3xi32>
-  %4 = "onnx.Add"(%2, %3) : (tensor<3xi32> , tensor<3xi32>) -> tensor<3xi32>
```

Comment:
this existing lit test had two uses of `%2` and therefore no longer triggers const prop with the new 1-use constraint

I edited the lit test to remove the dual uses and added a new lit test @test_add_const_associative2_2uses below to specifically test the dual use case